test/memory.test.js

```diff
@@ -19,82 +19,113 @@ describe('Memory connector', function () {
+    function createUserModel() {
+      var ds = new DataSource({
+        connector: 'memory',
+        file: file
+      });
 
-    var User = ds.createModel('User', {
-      name: String,
-      bio: String,
-      approved: Boolean,
-      joinedAt: Date,
-      age: Number
-    });
+      var User = ds.createModel('User', {
+        id: {
+          type: Number,
+          id: true,
+          generated: true
+        },
+        name: String,
+        bio: String,
+        approved: Boolean,
+        joinedAt: Date,
+        age: Number
+      });
+      return User;
+    }
 
-    var count = 0;
-    var ids = [];
-    async.eachSeries(['John1', 'John2', 'John3'], function (item, cb) {
-      User.create({name: item}, function (err, result) {
-        ids.push(result.id);
-        count++;
-        readModels(function (err, json) {
-          assert.equal(Object.keys(json.models.User).length, count);
-          cb(err);
+    it('should save to a json file', function(done) {
+      var User = createUserModel();
+      var count = 0;
+      var ids = [];
+      async.eachSeries(['John1', 'John2', 'John3'], function(item, cb) {
+        User.create({name: item}, function(err, result) {
+          ids.push(result.id);
+          count++;
+          readModels(function(err, json) {
+            assert.equal(Object.keys(json.models.User).length, count);
+            cb(err);
+          });
         });
-      });
-    }, function (err, results) {
-      // Now try to delete one
-      User.deleteById(ids[0], function (err) {
-        readModels(function (err, json) {
-          assert.equal(Object.keys(json.models.User).length, 2);
-          User.upsert({id: ids[1], name: 'John'}, function(err, result) {
-            readModels(function (err, json) {
-              assert.equal(Object.keys(json.models.User).length, 2);
-              var user = JSON.parse(json.models.User[ids[1]]);
-              assert.equal(user.name, 'John');
-              done();
+      }, function(err, results) {
+        if (err) {
+          return done(err);
+        }
+        // Now try to delete one
+        User.deleteById(ids[0], function(err) {
+          if (err) {
+            return done(err);
+          }
+          readModels(function(err, json) {
+            if (err) {
+              return done(err);
+            }
+            assert.equal(Object.keys(json.models.User).length, 2);
+            User.upsert({id: ids[1], name: 'John'}, function(err, result) {
+              if (err) {
+                return done(err);
+              }
+              readModels(function(err, json) {
+                if (err) {
+                  return done(err);
+                }
+                assert.equal(Object.keys(json.models.User).length, 2);
+                var user = JSON.parse(json.models.User[ids[1]]);
+                assert.equal(user.name, 'John');
+                assert(user.id === ids[1]);
+                User.update({id: user.id}, {name: 'John1'},
```

Comment:
This test is already too long and testing way too many things.

Don't add your new test code into this existing test as it makes things even worse. Please create a new test focused only on the use case you are fixing.